bench/benchlib_test.go

```diff
@@ -0,0 +1,221 @@
+package bench
+
+import (
+	"github.com/nats-io/nats"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	Byte    = 8
+	Million = 1000 * 1000
+)
+
+var baseTime = time.Now()
+
+func millionMessagesSecondSample(seconds int) *Sample {
+	messages := Million * seconds
+	start := baseTime
+	end := start.Add(time.Second * time.Duration(seconds))
+	nc := new(nats.Conn)
+
+	s := NewSample(messages, Byte, start, end, nc)
+	s.MsgCnt = uint64(messages)
+	s.MsgBytes = uint64(messages * Byte)
+	s.IOBytes = s.MsgBytes
+	return s
+}
+
+func TestDuration(t *testing.T) {
+	s := millionMessagesSecondSample(1)
+	duration := s.End.Sub(s.Start)
+	if duration != s.Duration() || duration != time.Second {
+		t.Fail()
```

Comment:
Would be nice to put some text and explain what failed, following a model such as `t.Fatalf("Expected %v, got %v", expectedValue, actualValue)`, when applicable of course.